### PR TITLE
Check go version and xgettext

### PR DESCRIPTION
**Pull request by AlekLefebvre**
Implements #833

### CODE REVIEW

File: tools/update-translator-sources [hunk 1]
```diff
@@ -9,11 +9,27 @@ set -u
...
   update_template_file
   update_translation_files
   go run ./tools/compile-translations.go
 }
 
+check_dependencies() {
+  go_version=`go version | { read _ _ v _; echo ${v#go}; }`
+  min_go_version="1.16"
+
+  if [ "$(printf '%s\n' "$min_go_version" "$go_version" | sort -V | head -n1)" = "$go_version" ]; then
```
Comment (strager) [inline tools/update-translator-sources:22]: This is a clever way to compare versions!

File: tools/update-translator-sources [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (strager) [inline tools/update-translator-sources:1]: Nit: This should be `#!/usr/bin/env bash`.

**Review by strager — CHANGES_REQUESTED** [review]
This check is broken if `go` is not installed. Here's the output I get:
```
$ ./tools/update-translator-sources
./tools/update-translator-sources: line 19: go: command not found
warning: Go version 1.16 or above required. You are using version
updating: po/messages.pot
........................................ done.
updating: po/de.po
........................................ done.
updating: po/en_US@snarky.po
........................................ done.
updating: po/fr_FR.po
........................................ done.
updating: po/pt_BR.po
........................................ done.
updating: po/sv_SE.po
........................................ done.
./tools/update-translator-sources: line 15: go: command not found
```

---

Must fix: This check is broken if `xgettext` is not installed. Here's the output I get:
```
$ ./tools/update-translator-sources
$ echo $?
127
```

**Comment by AlekLefebvre** [comment]
I fixed the issues and also fixed the error 127 when clang-format is not installed in tools/format.

**Review by strager — APPROVED** [review]

**Pull request merged**